src/client/js/partials/notFound.js

```diff
@@ -0,0 +1,19 @@
+if (window.history.length > 1) {
+  const backCta = document.getElementById('back-cta')
+  backCta.addEventListener('click', () => {
+    window.history.back()
+
+    // The call to `window.history.back()` might do nothing; for example, if the
+    // user has clicked Back to get to the 404 page: while there might not be
+    // another page to go back to, window.history.length will still be > 1
+    // because of the page the user can go forward to.
+    // Unfortunately there's no way to check whether it's possible to go back,
+    // or to check whether `.back()` succeeded, so we just use the fallback
+    // after half a second as passed:
+    setTimeout(() => {
+      document.location = '/'
+    }, 500)
```

Comment:
Question (non-blocking): Would it be possible to pass the prev path from the controller and append it to `#back-cta` via a `data-*` attribute?

Reply:
Oof. Ehm, maybe, but then we'd be tracking the user's navigation state on the server-side, which I think is a bit out of scope for this PR...

Reply:
Totally out of scope — just left this as a thought in case we need it in the future.